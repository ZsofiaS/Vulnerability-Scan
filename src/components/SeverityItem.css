.severity {
  flex-basis: 20%;
  list-style-type: none;
}
