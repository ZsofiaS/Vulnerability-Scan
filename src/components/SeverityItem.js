import React from 'react';
import '../styles/App.css';

const SeverityItem = (props) => {
  return (
    <li className='severityContainer'>
      <span>{props.severity}</span>
      <div
        className={props.count > 0 ? `severity ${props.severity}` : 'zero'}
        style={props.count < 10 ? {width: `${props.count}rem`} : {width: '10rem'}}>
        <span>{props.count}</span>
      </div>

    </li>
  )
}

export default SeverityItem;
