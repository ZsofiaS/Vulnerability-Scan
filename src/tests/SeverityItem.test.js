import React from 'react';
import { shallow, mount } from 'enzyme';
import SeverityItem from '../components/SeverityItem';
import toJson from 'enzyme-to-json';

const count = 2;

describe('renders SeverityItem component', () => {
  it('renders without crashing', () => {
    shallow(<SeverityItem />);
  });
});

describe('it passes props', () => {
  const severityItem = mount(<SeverityItem count={count} />);
  it('accepts props', () => {
    expect(severityItem.props().count).toEqual(2);
  });
});

describe('snapshot', () => {
  it('SeverityItem snapshot', () => {
    const tree = shallow(<SeverityItem />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
