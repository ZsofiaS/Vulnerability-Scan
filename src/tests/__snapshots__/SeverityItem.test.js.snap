// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SeverityItem snapshot 1`] = `
<li
  className="severityContainer"
>
  <span />
  <div
    className="zero"
    style={
      Object {
        "width": "10rem",
      }
    }
  >
    <span />
  </div>
</li>
`;
